ocs_ci/deployment/install_ocp_on_rhel.py

```diff
@@ -0,0 +1,211 @@
+"""
+This module will install OCP on RHEL nodes
+"""
+import logging
+import os
+
+from ocs_ci.framework import config
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.resources.pod import upload
+from ocs_ci.utility.templating import Templating
+from ocs_ci.utility.utils import (
+    create_rhelpod, get_cluster_version,
+)
+from semantic_version import Version
+
+
+logger = logging.getLogger(__name__)
+
+
+class OCPINSTALLRHEL(object):
+    """
+    Class to install OCP on RHEL nodes
+    """
+    def __init__(self, rhel_worker_nodes):
+        """
+        Initializes the required variables, create rhelpod, upload all the
+        helper files to pod and install required packages.
+
+        Args:
+            rhel_worker_nodes (list): list of RHEL nodes
+                e.g:['rhel1-0.vavuthu.qe.rh-ocs.com',
+                     'rhel1-1.vavuthu.qe.rh-ocs.com',
+                     'rhel1-2.vavuthu.qe.rh-ocs.com']
+
+        """
+        self.rhel_worker_nodes = rhel_worker_nodes
+        self.ssh_key_pem = config.DEPLOYMENT['ssh_key_private']
+        self.pod_ssh_key_pem = os.path.join(
+            constants.POD_UPLOADPATH,
+            self.ssh_key_pem.split("/")[-1]
+        )
+        self.ops_mirror_pem = os.path.join(
+            f"{constants.DATA_DIR}",
+            constants.OCP_PEM
+        )
+        self.cluster_path = config.ENV_DATA['cluster_path']
+        self.kubeconfig = os.path.join(
+            config.ENV_DATA['cluster_path'],
+            config.RUN.get('kubeconfig_location')
+        )
+        self.pod_name = "rhelpod"
+        self.pull_secret_path = os.path.join(
+            constants.TOP_DIR,
+            "data",
+            "pull-secret"
+        )
+        self.pod_pull_secret_path = os.path.join(
+            constants.POD_UPLOADPATH,
+            "pull-secret"
+        )
+        self.pod_kubeconfig_path = os.path.join(
+            constants.POD_UPLOADPATH,
+            config.RUN.get('kubeconfig_location').split("/")[-1]
+        )
+        self.rhelpod = create_rhelpod(
+            constants.DEFAULT_NAMESPACE,
+            self.pod_name
+        )
+
+        # map the cluster version with constants.REPO_MAPPING
+        cluster_version = get_cluster_version()
+        version_obj = Version(cluster_version)
+        version = ".".join(map(
+            str,
+            [version_obj.major, version_obj.minor, version_obj.patch]
+        ))
+        self.ocp_repo = constants.REPO_MAPPING[version]
```

Comment:
Do you think that we will change the repos each Z version?
If no, I would go with 4.2 only, if you think we will change between 4.2.Z then we can at least try to look for 4.2.Z in the map, if 4.2.Z won't be found, than try to directly access 4.2 (4.Y). So this will allow us to not have PR for each z-stream if not needed if we will keep 4.2 pointing to he latest one available.

I see that only one specific version is set in https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/repos/ocp_4_2.repo#L3  not sure if we can somehow associate this version in repo file with the version of OCP we have currently installed. Do you know more info about this? Thanks

Reply:
No. It won't change for Z version

Reply:
Please link the issue to cover what we have talked about in our today's mtg.